dashboard/app/controllers/browser_events_controller.rb

```diff
@@ -0,0 +1,69 @@
+# A controller for reporting browser logs and metrics to Cloudwatch
+
+class BrowserEventsController < ApplicationController
+  EXPERIMENT_FLAG_NAME = 'browser-cloudwatch-metrics'
+  FIREHOSE_STUDY_NAME = 'browser-cloudwatch-metrics-errors'
+
+  LOGS_CLIENT = Aws::CloudWatchLogs::Client.new
+  LOG_GROUP_NAME_PREFIX = 'browser-event-logs-'
+
+  before_action :check_preconditions
+
+  # POST /put_logs
+  def put_logs
+    body = JSON.parse(request.body.read)
+
+    return render status :bad_request, json: {message: 'missing required params: logs'} unless body["logs"]
+
+    logs = body["logs"].map do |log_payload|
+      {timestamp: (Time.now.to_f * 1000).to_i, message: decorate_log_payload(log_payload)}
+    end
+
+    resp = LOGS_CLIENT.put_log_events(
+      {
+        log_group_name: "#{LOG_GROUP_NAME_PREFIX}#{Rails.env}",
+        log_stream_name: Rails.env.to_s,
+        log_events: logs
+      }
+    )
+
+    if resp.rejected_log_events_info
+      fallback_log_to_firehose('rejected-log-events', resp.rejected_log_events_info)
+    end
+
+    render status: :ok, json: {}
+  rescue => exception
+    fallback_log_to_firehose('put-log-events-error', {exception: exception, logs: logs})
```

Comment:
Is the fallback to firehose necessary?

Reply:
I added this for the experiment phase, in case we find some issue with how we're logging to cloudwatch. Longer term though, I don't think we'd need to do this.

Reply:
why not adding a honeybadger instead of a firehose event? You can add the log as "context" and honeybadger will give you more for debugging than this firehose event.

Reply:
Good question, I was considering using Honeybadger here in general initially but discussed with @davidsbailey and it seemed like Firehose would be a better fit. However, re-reading that [thread](https://codedotorg.slack.com/archives/C0T10HG6N/p1680637337172519) I think Honeybadger may make more sense for this exception case specifically - what do you think?

Reply:
oh I see. One is not technically an error but debugging.

Things i would consider:
1) Keep it consistent (Honeybadger) since it is an experiment. I understand that it is technically not an error, but it is something to be "aware" or "alert" from backend perspective.
2) This is the irony of this logger. You are meta thinking. If the logger fails then where do we log?

There is no clear answer here, it is about perspective. I agree with Dave's comment on using firehose for a 200.

Tbh, since this is in experiment phase, If I were you I would keep it consistent in honeybadger since you do want it for debugging

Reply:
let me know what you decide! I trust you on this, you are being pulled in different directions which are both valid :)

Reply:
What action should be taken when these failures occur?

With Firehose, we're not even going to know about it unless someone goes to investigate this specific thing.

With Honeybadger it goes into the stack of errors to process and we get free metrics about the frequency of the error.

Honeybadger gets my vote. The system did not do what it was supposed to do, that's an error in my book.

Reply:
That makes sense! HB feels like the right option for when a failure occurs, while Firehose feels right to report about unintended side effects (e.g. logs being rejected) even though the service is returning 200.

Reply:
Updated to report errors to Honeybadger